compliance/virtualmachines/relay/testutils/testutils.go

```diff
@@ -0,0 +1,120 @@
+package relaytest
+
+import (
+	"context"
+	"io"
+	"net"
+	"os"
+	"time"
+
+	"github.com/mdlayher/vsock"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	v1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+// MockVsockConn implements net.Conn and allows tests to craft vsock-backed connections easily.
+type MockVsockConn struct {
+	data         []byte
+	delay        time.Duration
+	remoteAddr   net.Addr
+	readDeadline time.Time
+}
+
+func NewMockVsockConn() *MockVsockConn {
```

Comment:
nit: make sure the mock can only be called from test code
```suggestion
func NewMockVsockConn(_ testing.TB) *MockVsockConn {
```